AccessibilityInsightsForAndroidService/app/src/main/java/com/microsoft/accessibilityinsightsforandroidservice/FocusElementHighlight.java

```diff
@@ -0,0 +1,152 @@
+import java.util.HashMap;
+
+public class FocusElementHighlight extends View {
+  private static final String TAG = "FocusElementHighlight";
+  private AccessibilityNodeInfo eventSource;
+  private int yOffset;
+  private int tabStopCount;
+  private boolean isCurrent;
+  private boolean isReset;
+  private int radius;
+  private int xCoordinate;
+  private int yCoordinate;
+  // performance based variables
+  private HashMap<String, Paint> currentPaints;
+  private HashMap<String, Paint> nonCurrentPaints;
+  private Rect rect;
+
+  public FocusElementHighlight(Context context){
+    super(context);
+  }
+  public FocusElementHighlight(Context context, AttributeSet attrs){
+    super(context, attrs);
+  }
+  public FocusElementHighlight(Context context, AttributeSet attrs, int toolInt){
+    super(context, attrs, toolInt);
+  }
+  public FocusElementHighlight(
+      Context context,
+      AccessibilityNodeInfo eventSource,
+      HashMap<String, Paint> currentPaints,
+      HashMap<String, Paint> nonCurrentPaints,
+      int radius,
+      int tabStopCount,
+      boolean isCurrent) {
+    super(context);
+    this.eventSource = eventSource;
+    this.yOffset = getYOffset();
+    this.tabStopCount = tabStopCount;
+    this.currentPaints = currentPaints;
+    this.nonCurrentPaints = nonCurrentPaints;
+    this.radius = radius;
+    this.isCurrent = isCurrent;
+    this.isReset = false;
+    this.rect = new Rect();
+  }
+
+  @Override
+  protected void onDraw(Canvas canvas) {
+    super.onDraw(canvas);
+
+    if (eventSource == null) {
+      return;
+    }
+
+    eventSource.getBoundsInScreen(rect);
+    rect.offset(0, yOffset);
+
+    this.xCoordinate = rect.centerX();
+    this.yCoordinate = rect.centerY();
+
+    if (isReset) {
+      this.drawInnerCircle(
+          xCoordinate, yCoordinate, radius, nonCurrentPaints.get("transparent"), canvas);
+      this.drawNumberInCircle(
+          xCoordinate,
+          yCoordinate,
+          tabStopCount,
+          nonCurrentPaints.get("transparent"),
+          canvas);
+      this.drawOuterCircle(
+          xCoordinate, yCoordinate, radius, nonCurrentPaints.get("transparent"), canvas);
+      return;
+    }
+
+    if (!isCurrent) {
+      this.drawInnerCircle(
+          xCoordinate, yCoordinate, radius, nonCurrentPaints.get("innerCircle"), canvas);
+      this.drawNumberInCircle(
+          xCoordinate, yCoordinate, tabStopCount, nonCurrentPaints.get("number"), canvas);
+      this.drawOuterCircle(
+          xCoordinate, yCoordinate, radius, nonCurrentPaints.get("outerCircle"), canvas);
+      return;
+    }
+
+    this.drawInnerCircle(
+        xCoordinate, yCoordinate, radius, currentPaints.get("innerCircle"), canvas);
+    this.drawNumberInCircle(
+        xCoordinate, yCoordinate, tabStopCount, currentPaints.get("number"), canvas);
+    this.drawOuterCircle(
+        xCoordinate, yCoordinate, radius, currentPaints.get("outerCircle"), canvas);
+  }
+
+  public int getYOffset() {
```

Comment:
This method and the ones below that are only called in this class should likely be private.

Same also applies to FocusElementLine.